### DISCUSSION TITLE
Should we use ptPixelLocationRaw to replace ptPixelLocation in WM_Pointer?

### DISCUSSION THREAD

**Discussion by lindexi** [AvaloniaUI/Avalonia#16848] [General]
As we can see, the Avalonia will use ptPixelLocation in AppWndProc

https://github.com/AvaloniaUI/Avalonia/blob/de86c5a058915fca10adae4fc66b3e3ac8a8df86/src/Windows/Avalonia.Win32/WindowImpl.AppWndProc.cs#L1055

As the document https://learn.microsoft.com/en-us/windows/win32/api/winuser/ns-winuser-pointer_info says, the ptPixelLocation is the predicted screen coordinates of the pointer, in pixels.

And I do not think we can use the predicted pointer here.

And the WPF use the ptPixelLocationRaw, see https://github.com/dotnet/wpf/blob/a13bb634c1dcdf0bf7cda9a02c448b53366b8550/src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Input/Stylus/Pointer/PointerStylusDevice.cs#L637

```csharp
                Point rawScreenPoint = new Point(_pointerData.Info.ptPixelLocationRaw.X, _pointerData.Info.ptPixelLocationRaw.Y);
                Point ptDevice = PointUtil.ScreenToClient(rawScreenPoint, _inputSource.Value);
```

**Comment by lindexi** [comment]
Imagine if you used the points from predicted for handwriting drawing, that the stroke will be full of serrations, not a smooth curve.

Reference https://github.com/AvaloniaUI/Avalonia/issues/1477

**Reply by maxkatz6** [reply]
According to the documentation, prediction only works with PT_TOUCH (not PT_PEN or PT_MOUSE), and to "compensate for visual lag due to inherent delays in sensing and processing the pointer location on the digitizer".
While I am not against switching to RAW values, it doesn't sound as an unwanted behavior here.
